### PR TITLE
clean up

### PR DESCRIPTION
**Pull request by cqc-melf**

### CODE REVIEW

**Comment by cqc-melf** [comment]
I think it would be fine to keep the _metadata.py because we need the version written down for the doc build anyway? Any objections?

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -206,12 +206,12 @@ jobs:
...
-        #  with:
-        #    user: __token__
-        #    password: ${{ secrets.PYPI_EXTENSIONS_API_TOKEN }}
-        #    verbose: true
+    - name: Publish wheels
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_PYTKET_QUANTINUUM_API_TOKEN }}
+        verbose: true
```
Comment (cqc-alec) [inline .github/workflows/build_and_test.yml:215]: Can we combine the Linux, MacOS and Windows jobs into a single job using `matrix` for the OS? (For an example, see e.g. the `test_libs.yml` in the tket repo.) (We can still perform steps like the mypy check conditionally on the OS.)

**Review by cqc-alec — CHANGES_REQUESTED** [review]
- Needs catching up with latest changes from the pytket-extensions repo.
- I don't think `base-test-requirements.txt` is needed any more; we can just use `test-requirements.txt`.
- Can probably remove some filters from `pytest.ini`.
- Need new tokens and secrets for `issue.yml`?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -206,12 +206,12 @@ jobs:
...
 
   docs:
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply .github/workflows/build_and_test.yml:97]: I have done that and combined that to one step.

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -45,15 +46,25 @@ jobs:
...
+    - name: Build and test including remote checks (3.9) mypy
+      if: |
+        ((github.event_name == 'pull_request' &&  github.event.pull_request.head.repo.full_name == github.repository)
+        || github.event_name == 'release'
+        || contains(github.ref, 'refs/heads/wheel')) && ${{ matrix.os }} == 'macos-11'
+      run: |
+        ./.github/workflows/build-test mypy
+      env:
+        PYTKET_RUN_REMOTE_TESTS: 1
+        MODULE: ${{ matrix.module }}
```
Comment (cqc-alec) [inline .github/workflows/build_and_test.yml:58]: There isn't a `matrix.module` is there?

**Review by cqc-alec — CHANGES_REQUESTED** [review]

**Comment by cqc-alec** [comment]
> I think it would be fine to keep the _metadata.py because we need the version written down for the doc build anyway? Any objections?

OK

File: .github/workflows/build_and_test.yml (hunk 3 above)
Comment (cqc-melf) [reply .github/workflows/build_and_test.yml]: The env variable was unused, have removed that.

File: .github/workflows/build-test [hunk 4]
```diff
@@ -27,92 +27,34 @@ git clean -dfx
...
-        if [[ "${MODULE}" = "pytket-qsharp" && "${PLAT}" != "Darwin" ]]
-        then
-            dotnet iqsharp install --user
-        fi
-
-        # Disable the remote tests for pytket-aqt and -qsharp for now
-        if [[ "${MODULE}" = "pytket-aqt" || "${MODULE}" = "pytket-qsharp" ]]; then
-            unset PYTKET_RUN_REMOTE_TESTS; pytest --doctest-modules
-        else
-            pytest --doctest-modules
```
Comment (cqc-alec) [inline .github/workflows/build-test:110]: Oops, deleted line to run tests!

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: setup.py [hunk 5]
```diff
@@ -41,12 +41,13 @@
...
     include_package_data=True,
     install_requires=[
-        "pytket ~= 1.3",
+        "pytket ~= 1.4",
         "requests >= 2.2",
         "types-requests",
         "websockets >= 7.0",
         "nest_asyncio >= 1.2",
         "pyjwt ~= 2.4",
+        "msal ~= 1.18",
```
Comment (cqc-alec) [inline setup.py:50]: What is this for?

File: _metadata.py [hunk 6]
```diff
@@ -1,2 +1,2 @@
-__extension_version__ = "0.5.0rc0"
+__extension_version__ = "0.5.0"
```
Comment (cqc-alec) [inline _metadata.py:1]: It would probably make sense to bump the version tp 0.6.0.

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 7]
```diff
@@ -209,7 +208,7 @@ def _dict_to_backendinfo(cls, dct: Dict[str, Any]) -> BackendInfo:
         return BackendInfo(
             name=cls.__name__,
             device_name=name,
-            version=__extension_version__,
+            version="0.5.0",
```
Comment (cqc-alec) [inline pytket/extensions/quantinuum/backends/quantinuum.py:211]: Why are we hard-coding this here?

File: .github/workflows/build_and_test.yml [hunk 8]
```diff
@@ -38,23 +39,29 @@ jobs:
...
         ./.github/workflows/build-test nomypy
-      env:
-        MODULE: ${{ matrix.module }}
     - name: Set up Python 3.9
       if: github.event_name == 'pull_request' || github.event_name == 'release' || contains(github.ref, 'refs/heads/wheel')
       uses: actions/setup-python@v3
       with:
         python-version: '3.9'
-    - name: Build and test including remote checks (3.9)
+    - name: Build and test including remote checks (3.9) mypy
```
Comment (cqc-alec) [inline .github/workflows/build_and_test.yml:47]: Looks like both the mypy and nomypy versions have been run on all platforms?

File: setup.py (hunk 5 above)
Comment (cqc-melf) [reply setup.py:50]: It is for the Microsoft login. (added with the other MFA stuff)

File: pytket/extensions/quantinuum/backends/quantinuum.py (hunk 7 above)
Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/quantinuum.py]: Oh sorry, I forgot to change that again!

File: _metadata.py (hunk 6 above)
Comment (cqc-melf) [reply _metadata.py]: updated in the @cqc-melf
[fix workflow and version in tests](https://github.com/CQCL/pytket-quantinuum/pull/3/commits/c3b9fa3e6f4c30aae59efb1492b6e80d9c5f101a)

File: .github/workflows/build_and_test.yml (hunk 8 above)
Comment (cqc-melf) [reply .github/workflows/build_and_test.yml:47]: Thank you, should be fixed now. @cqc-melf
[fix workflow and version in tests](https://github.com/CQCL/pytket-quantinuum/pull/3/commits/c3b9fa3e6f4c30aae59efb1492b6e80d9c5f101a)

File: .github/workflows/build_and_test.yml [hunk 9]
```diff
@@ -38,23 +39,23 @@ jobs:
...
         python-version: '3.9'
-    - name: Build and test including remote checks (3.9)
-      if: |
-        (github.event_name == 'pull_request' &&  github.event.pull_request.head.repo.full_name == github.repository)
-        || github.event_name == 'release'
-        || contains(github.ref, 'refs/heads/wheel')
+    - name: Build and test including remote checks (3.9) mypy
+      if:  (matrix.os == 'macos-11') && ((github.event_name == 'pull_request' &&  github.event.pull_request.head.repo.full_name == github.repository) || github.event_name == 'release' || contains(github.ref, 'refs/heads/wheel'))
       run: |
         ./.github/workflows/build-test nomypy
```
Comment (cqc-alec) [inline .github/workflows/build_and_test.yml:50]:
```suggestion
        ./.github/workflows/build-test mypy
```

**Review by cqc-alec — CHANGES_REQUESTED** [review]

**Comment by cqc-melf** [comment]
Thank you for spotting all the stupid errors I make, @cqc-alec !

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**